File: wow-commons/src/main/java/wow/commons/model/professions/ProfessionSpecialization.java
package wow.commons.model.professions;

import lombok.AllArgsConstructor;
import lombok.Getter;
import wow.commons.util.EnumUtil;

/**
 * User: POlszewski
 * Date: 2022-10-31
 */
@AllArgsConstructor
@Getter
public enum ProfessionSpecialization {
	GNOMISH_ENGINEER("Gnomish Engineer", Profession.ENGINEERING),
	GOBLIN_ENGINEER("Goblin Engineer", Profession.ENGINEERING),
	MASTER_SWORDSMITH("Master Swordsmith", Profession.BLACKSMITHING),
	MOONCLOTH_TAILORING("Mooncloth Tailoring", Profession.TAILORING),
	SHADOWEAVE_TAILORING("Shadoweave Tailoring", Profession.TAILORING),
	SPELLFIRE_TAILORING("Spellfire Tailoring", Profession.TAILORING);

	private final String key;
	private final Profession profession;

	public static ProfessionSpecialization parse(String value) {
		return EnumUtil.parse(value, values(), x -> x.key);
	}

	public static ProfessionSpecialization tryParse(String value) {
		return EnumUtil.tryParse(value, values(), x -> x.key);
	}

	@Override
	public String toString() {
		return key;
	}
}

File: apps/wow-minmax-ui/src/app/modules/shared/model/character/ProfessionSpecialization.ts
import { Profession } from "./Profession";

export enum ProfessionSpecialization {
	GNOMISH_ENGINEER = 'GNOMISH_ENGINEER',
	GOBLIN_ENGINEER = 'GOBLIN_ENGINEER',
	MASTER_SWORDSMITH = 'MASTER_SWORDSMITH',
	MOONCLOTH_TAILORING = 'MOONCLOTH_TAILORING',
	SHADOWEAVE_TAILORING = 'SHADOWEAVE_TAILORING',
	SPELLFIRE_TAILORING = 'SPELLFIRE_TAILORING',
}

export function getProfessionSpecializations(profession?: Profession) {
	if (!profession) {
		return [];
	}
	if (profession === Profession.TAILORING) {
		return [
			ProfessionSpecialization.MOONCLOTH_TAILORING,
			ProfessionSpecialization.SHADOWEAVE_TAILORING,
			ProfessionSpecialization.SPELLFIRE_TAILORING,
		];
	}
	if (profession === Profession.ENGINEERING) {
		return [
			ProfessionSpecialization.GNOMISH_ENGINEER,
			ProfessionSpecialization.GOBLIN_ENGINEER,
			ProfessionSpecialization.MASTER_SWORDSMITH,
		];
	}
	return [];
}
